kube-client/src/client/mod.rs

```diff
@@ -238,15 +237,18 @@ impl Client {
         Ok(text)
     }
 
-    /// Perform a raw HTTP request against the API and get back the response
-    /// as a stream of bytes
-    pub async fn request_text_stream(
-        &self,
-        request: Request<Vec<u8>>,
-    ) -> Result<impl Stream<Item = Result<Bytes>>> {
+    /// Perform a raw HTTP request against the API, get back the response
+    /// and return it as an object that implements [`AsyncBufRead`] and thus
+    /// [`AsyncBufReadExt`]. Users can call `.lines()` on it to get a newline
+    /// buffered stream of logs.
+    pub async fn request_text_stream(&self, request: Request<Vec<u8>>) -> Result<impl AsyncBufRead> {
```

Comment:
Since this is already a breaking change (and there is a high chance nobody is actually using this directly), I would take the chance to rename this to `request_stream`, this now streams the raw response, wether it is text or some binary.

@clux @nightkr opinions?